File: leetcode/exercises/2.easy.two_sum/two_sum_1.py
# -----------------------
# Created       : 20/10/2025
# Last Edited   : 20/10/2025
# Self Solved   : False
# Big O         : O(n)
# Topics        : 
# -----------------------

class Solution:
	def twoSum(self, nums: list[int], target: int) -> list[int]:
		map = {}
		for index,value in enumerate(nums):
			complement = target - value
			if complement in map:
				return [index, map[complement]]
			map[value] = index

sol = Solution()
print(sol.twoSum([1,5,6,2,2,8], 11))

File: leetcode/exercises/2.easy.two_sum/two_sum_1.go
package main
import "fmt"

func twoSum(nums []int, target int) []int {
    m := map[int]int{}
    for i, v := range nums {
        complement := target - v
        _, ok := m[complement] 
        if ok { return []int{m[complement], i} }
        m[v] = i
    }
    return []int{0,0}
}

func main() {
    fmt.Println(twoSum([]int{1,5,6,2,2,8}, 11))
}
